packages/reactivity/src/baseHandlers.ts

```diff
@@ -120,11 +120,11 @@ function createGetter(isReadonly = false, shallow = false) {
     }
 
     if (isRef(res)) {
-      // ref unwrapping - does not apply for Array + integer key.
-      const shouldUnwrap = !targetIsArray || !isIntegerKey(key)
-      return shouldUnwrap ? res.value : res
+      // ref unwrapping - ignore for Array + integer key.
+      const isIgnore = targetIsArray && isIntegerKey(key)
+      return isIgnore ? res : res.value
```

Comment:
Could also be
```suggestion
      return (targetIsArray && isIntegerKey(key)) ? res : res.value
```

make sure to run the format script after doing these changes

Reply:
`npm run format `  Changed other files
Whether should add an automatic task to execute npm format